### PR TITLE
Feat/price refactoring

### PR DESCRIPTION
**Pull request by BenDanielGreep**
I'm opening this PR without the currency separator code being written.
However the other aspects of the code work well.
The currency separator will be put aside for this sprint, due to other priorities. 

## Objective
What: Create a more dynamic Price component
Why: to more easily show the correct price format, including sale price when it exists
How: Refactor the existing component, adding logic and creating new items in the configuration
Usage: homepage, srp. pdp. refinement list price slider
## Type

- [ ] Bug Fix
- [ ] New Feature
- [X] Performance Tweaks
- [ ] Style Tweaks
- [X] Code Refactoring
- [ ] Documentation
- [ ] Tests

## Tested

localhost

## Documented

- [X] Have you documented in changed file using comments
- [ ] Have you added instructions to the README if needed?

### CODE REVIEW

**Comment by BenDanielGreep** [comment]
> In my opinion I don't know if you'll do this but I'll integrate a currency switcher in the header like Hugo Boss on their website because maybe I speak english so i want english language, but I living in Paris so I want the price in euros. Screenshot 2022-08-11 at 17 30 54 screencapture-hugoboss-uk-men-2022-08-11-17_34_10

Great suggestion @VirAlex ! Do you mean that you're going to do it on a separate branch?

**Comment by VirAlex** [comment]
> > In my opinion I don't know if you'll do this but I'll integrate a currency switcher in the header like Hugo Boss on their website because maybe I speak english so i want english language, but I living in Paris so I want the price in euros. Screenshot 2022-08-11 at 17 30 54 screencapture-hugoboss-uk-men-2022-08-11-17_34_10
> 
> Great suggestion @VirAlex ! Do you mean that you're going to do it on a separate branch?

👋  no it was just an improvement you could have did on your branch, but you can create a new issue on it. 
NP 😉

File: src/config/hitsConfig.js [hunk 1]
```diff
@@ -22,7 +22,9 @@ export const hitsConfig = {
   genderFilter: 'genderFilter',
   hierarchicalCategories: 'hierarchicalCategories',
   sizeFilter: 'sizeFilter',
-  price: 'unformated_price',
+  price: 'price',
+  onSale: 'onSale',
```
[the remark this answers is not in the document]
Comment (BenDanielGreep) [reply src/config/hitsConfig.js:26]: So if I rename onSale to isOnSale, then I can't use isOnSale as a var in the PriceBuilder component. Any ideaS?

Screenshot 2022-08-16 at 14 52 07

File: src/components/hits/components/Price.jsx [hunk 2]
```diff
@@ -8,49 +8,80 @@ import { hitsConfig } from '@/config/hitsConfig';
...
-          {displayCurrency && currencySymbol}
-        </>
-      );
-    default:
-      return (
-        <>
-          {displayCurrency && currencySymbol}
+  // Check if the currency should be on the right, ie: if it is Euros.
+  // You can add other currencies to this function if you want them on the right of your price
+  const isCurrencyRight = '€' === currencySymbol;
```
[the remark this answers is not in the document]
Comment (BenDanielGreep) [reply src/components/hits/components/Price.jsx]: Fixed

File: src/components/hits/components/Price.jsx [hunk 3]
```diff
@@ -8,49 +8,80 @@ import { hitsConfig } from '@/config/hitsConfig';
 
 import get from 'lodash/get';
 
+import {
+  currencySymbolAtom,
+  shouldDisplayCurrency,
+  // shouldInvertPriceSeparators,
```
[the remark this answers is not in the document]
Comment (BenDanielGreep) [reply src/components/hits/components/Price.jsx:15]: Fixed

**Review by matthew1809 — APPROVED** [review]
GG!

**Pull request merged**